accounting/transferwise/models.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2017-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+TransferWise models to be used by the Accounting service.
+"""
+
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from accounting.account.models import Account
+from accounting.common.models import CommonModel, UuidModel
+from accounting.invoice.models import Invoice
+from accounting.third_party_api.google.mixins import GoogleDriveMixin
+from accounting.transferwise.mixins import TransferWiseCsvMixin
+
+
+class TransferWiseBulkPayment(CommonModel, UuidModel, TransferWiseCsvMixin, GoogleDriveMixin):
+    """
+    A model that represents TransferWise bulk payments.
+
+    Can be used to generate bulk payment CSVs.
+    """
+
+    date = models.DateTimeField(
+        default=timezone.now,
+        help_text=_("The date this bulk payment was created and uploaded to TransferWise. "
+                    "Defaults to right now, but can be changed."))
+    sender = models.ForeignKey(
```

Comment:
We don't have a use case where the sender is not OpenCraft, right? So I don't think we need this field, especially one that we have to select every time we create a bulk payment. Can this be replaced by a global config variable, or something similar that isn't set on every bulkpayment object?

Reply:
@antoviaque If we do this the codebase will be frankenstein: in some places, like with invoices, we'll have generic clients, and here we'll have a pre-selected one. If we decide to undo this we should only do it if we undo the invoice stuff too, which is now going to be a lot of work since it's quite widespread.

But I think you don't need to worry about the manual work here: given my work in #10, I can easily create one extra job, to run on the 6th, that automatically makes this bulk payment CSV and assigns OpenCraft properly through the new `BILLING_CYCLE_USERS` setting, which is basically what you're talking about.

Reply:
@UmanShahzad This would be specific to the TransferWise payments. It can make sense to have different providers set for invoices, as you have both incoming and outgoing invoices. But we are only going to create TransferWise bulk payments for the payments we make. And even if it was a frankenstein, I would still prefer to not merge code like this, and instead log a ticket to undo it in the rest of the code base.

Reply:
@antoviaque I find it a little unsettling that we'd make a ticket to undo generic code to be un-generic -- that's the first I've heard of someone wanting that. This repository is open source code, so it's counterproductive to be doing this, considering you *won't have to set it yourself* given upcoming work.

Reply:
@UmanShahzad I didn't ask to make the rest of the code ungeneric - only to remove the provider field in this specific model. And making things less generic can remove complexity - so yes, it can also be a good thing.

Reply:
@antoviaque I understand you meant only this `sender` field. If we're going to delete this though, it'd honestly only make sense in a separate ticket that does it together with the `client` field on `Invoice`, to keep the codebase consistent in one go. Otherwise it won't reduce complexity, it'll just make it harder to understand how the pieces fit if reading from scratch. Even *I'd* be confused by it, as the person writing this stuff.

Reply:
@antoviaque Made https://tasks.opencraft.com/browse/OC-3722 for this.